org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/IFilter.java

```diff
@@ -23,12 +23,17 @@
 	 * This method is called for every method. The filter implementation is
 	 * expected to inspect the provided method and report its result to the
 	 * given {@link IFilterOutput} instance.
-	 * 
+	 *
+	 * @param className
+	 *            class name
+	 * @param superClassName
+	 *            superclass name
 	 * @param methodNode
 	 *            method to inspect
 	 * @param output
 	 *            callback to report filtering results to
 	 */
-	void filter(MethodNode methodNode, IFilterOutput output);
+	void filter(String className, String superClassName, MethodNode methodNode,
```

Comment:
@Godin We might need to add more params in the future here. What do you think about a new interface `IFilterContext` where we pass a instance here?